### ISSUE TITLE
opening transferred files - strange behaviour

### ISSUE THREAD

**Issue by m-wieczorek** [qTox/qTox#1325]
[lubuntu 14.04 i386]

While I'm opening received files both buttons 

![zrzut ekranu z 2015-03-02 17 07 20](https://cloud.githubusercontent.com/assets/11274280/6445011/afce04f2-c101-11e4-8702-c0740eb88cad.png)

work exactly as they are supposed to - the upper one opens a file, the second opens file location with default filemanger (PCManFM).

However when I'm trying to do the same with the files I've just sent to somebody, the buttons start to act weirdly - the upper one is trying to open file in a system default web browser and the lower one is starting audacious and generally acts like if I was trying to open some kind of octet-stream.

I know it might my system issue, however I've got no idea what might be the cause.

**Comment by zetok** [comment]
With what files did you try?

**Comment by m-wieczorek** [comment]
Oh, I forgot to mention the file type - those were images - .png

**Comment by TheLastProject** [comment]
Well, I tested it myself with .png images, but couldn't reproduce this.

Is it a specific image causing this weird behaviour, or just any .png image? Also, did you send and receive the same file? I'm starting to think you may have a .png file that is not truly a .png file, perhaps.

**Comment by m-wieczorek** [comment]
So I've tested it today with 3 images created in gimp (just to be sure they are really images):

test.png
test2.jpg
test3.gif

I sent them and asked to send them back to me and everything works fine apart from the fact that .png file is still being opened in the private session of firefox (I didn't mention about this session being private before) and it concerns received as well as sent files.

Maybe the issue with opening the previous files in their locations was connected with their names pattern - there were screenshots (created with scrot) with the file names like:

2015-03-01-172313_1280x800_scrot.png

and so on.

So to sum up, I still don't know why qtox opens some .png files in private session of firefox.
Outside of qtox the .png images are opened correctly (with GPicView).

I'm adding here my test.png image:
![test](https://cloud.githubusercontent.com/assets/11274280/6468721/4ba3008c-c1d7-11e4-8e3e-758dd6fa0ee9.png)

**Comment by m-wieczorek** [comment]
Well @TheLastProject I suppose you were right - when I renamed test.png to 2015-03-01-192412_1280x800_scrot.png nothing has changed. However it is still opened in firefox private mode. I assume that indeed these scrot screenshots aren't really .png images.

**Comment by dubslow** [comment]
And this only happens on sending files?

@krepa098

**Comment by m-wieczorek** [comment]
@dubslow it happens when I'm trying to open just received or sent .png files using the upper button: 
![button](https://cloud.githubusercontent.com/assets/11274280/6484418/0fad5a14-c27c-11e4-9b7f-1c32048b6599.png)

**Comment by TheLastProject** [comment]
I can't reproduce this, not even with the supplied image. At this point, I'm assuming Firefox's private mode is simply set as the default for png images.

After all, this is the only code:
`QDesktopServices::openUrl(QUrl("file://" + file.filePath(), QUrl::TolerantMode));`

The documentation says:
[If the URL is a reference to a local file (i.e., the URL scheme is "file") then it will be opened with a suitable application instead of a Web browser.](http://qt-project.org/doc/qt-4.8/qdesktopservices.html#openUrl).

**Comment by zetok** [comment]
Couldn't reproduce. Closing, since it looks like an issue with your system configuration.
